### PR TITLE
Delete message from chat

### PR DESCRIPTION
**Pull request by kngako**
https://github.com/stakwork/sphinx-kotlin/issues/196

### CODE REVIEW

File: sphinx/application/data/features/feature-repository/src/main/java/chat/sphinx/feature_repository/SphinxRepository.kt [hunk 1]
```diff
@@ -1598,6 +1599,40 @@ abstract class SphinxRepository(
...
+                    }
+                }
+            } else {
+                networkQueryMessage.deleteMessage(message.id).collect { loadResponse ->
+                    @Exhaustive
+                    when (loadResponse) {
+                        is LoadResponse.Loading -> {}
+                        is Response.Error -> {
+                            LOG.e(TAG, loadResponse.message, loadResponse.exception)
+                            // TODO: Give user some information about failure maybe?
```
Comment (kevkevinpal) [inline sphinx/application/data/features/feature-repository/src/main/java/chat/sphinx/feature_repository/SphinxRepository.kt:1619]: Might be able to do some sort of modal or something here indicating failure?

**Comment by kevkevinpal** [comment]
Also not sure if we should be merging to the develop branch or not

**Comment by tomastiminskas** [comment]
@kngako a few changes required:
- Make message options menu a bit wider since the "Delete message" text is pretty close to the right edge
- Get the response of the deleteMessage endpoint on `ChatViewModel` and use `ChatSideEffect.Notify` to show an error toast to the user
- On `DashboardChat` -> `getMessageText` check for deleted status and show "Message deleted" instead of message text. That is the text that shows the last message on Dashboard rows

**Comment by kngako** [comment]
Thanks for the feedback @kevkevinpal and @tomastiminskas. I've applied the changes and updated the pull request to use develop as the base.

**Comment by tomastiminskas** [comment]
@kngako last changes:
- "message deleted" should be replaced by "Message deleted"
- On the `getMessageText()` of `DashboardChat`, the check for deleted status shouldn't be inside the Message type statement. Since it should apply to other messages types like attachments. It should be placed after `messageDecryptionError` and before `type.isMessage()`

Once that's done I think we can merge this.

**Comment by kngako** [comment]
@tomastiminskas cool. I just made the change.

File: sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatFragment.kt [hunk 2]
```diff
@@ -370,7 +370,9 @@ abstract class ChatFragment<
                                 viewModel.copyMessageText(holderState.message)
                             }
                             is MenuItemState.Delete -> {
-                                // TODO: Implement
```
Comment (05nelsonm) [inline sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatFragment.kt:373]: Should always prefer to utilize viewModelScope in possible, as cancellation of the lifecycleScope will cancel the suspend function being called, thus not handling the returned value.

```kotlin
// from screen
viewModel.deleteMessage(holderState.message)

// from viewmodel
fun deleteMessage(message: Message) {
    viewModelScope.launch(mainImmediate) {
        when (messageRepository.deleteMessage(message)) {
            // ......
        }
    }
}
```

**Review by 05nelsonm — CHANGES_REQUESTED** [review]

**Pull request merged**